### PR TITLE
feat(plasma-tokens): Refactoring tokens packages

### PR DESCRIPTION
**Pull request by neretin-trike**
Удалил не используемые пакеты `plasma-tokens-android` и `plasma-tokens-ios-swift`, поскольку они не содержат никакой логики.

Удалил токены размеров (`sizes`) из общего набора токенов темы и перенёс эти значения в компоненты `Button` и `ActionButton`.

Удалил функционал генерации темы с помощью `cli`, т.к. он дублируется в веб-сервисе plasma-theme-builder. 

Удалил функционал генерации файла темы `theme.json`, который никак не используется.

Произвёл рефакторинг библиотеки `@salutejs/plasma-tokens`. Теперь структура файлов выглядит так:

image

Добавил обработку префиксов (например `plasma-colors`) для старых токенов, чтобы новые темы частично работали с компонентами:

image

image

image

Обновил директорию `src` после всех произведенных изменений.

:baby_chick: Download canary assets:

[colors--canary.487.4794934885.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.487.4794934885.xml)  
[color_sberHealth_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.487.4794934885.swift)  
[color_sberHealth_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.487.4794934885.kt)  
[color_sberHealth_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.487.4794934885.ts)  
[color_sbermarket_business_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.487.4794934885.swift)  
[color_sbermarket_business_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.487.4794934885.kt)  
[color_sbermarket_business_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.487.4794934885.ts)  
[color_sbermarket_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.487.4794934885.swift)  
[color_sbermarket_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.487.4794934885.kt)  
[color_sbermarket_metro_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.487.4794934885.swift)  
[color_sbermarket_metro_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.487.4794934885.kt)  
[color_sbermarket_metro_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.487.4794934885.ts)  
[color_sbermarket_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.487.4794934885.ts)  
[color_sbermarket_selgros_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.487.4794934885.swift)  
[color_sbermarket_selgros_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.487.4794934885.kt)  
[color_sbermarket_selgros_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.487.4794934885.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.487.4794934885.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.487.4794934885.kt)  
[color_sbermarket_wlbusiness_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.487.4794934885.ts)  
[color_sberprime_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.487.4794934885.swift)  
[color_sberprime_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.487.4794934885.kt)  
[color_sberprime_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.487.4794934885.ts)  
[PlasmaTokensColor--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.487.4794934885.swift)  
[shadow_sbermarket_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.487.4794934885.ts)  
[typo_mage_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.487.4794934885.swift)  
[typo_mage_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.487.4794934885.kt)  
[typo_mage_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.487.4794934885.ts)  
[typo_plasma_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.487.4794934885.swift)  
[typo_plasma_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.487.4794934885.kt)  
[typo_plasma_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.487.4794934885.ts)  
[typo_ruler_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.487.4794934885.swift)  
[typo_ruler_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.487.4794934885.kt)  
[typo_ruler_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.487.4794934885.ts)  
[typo_sage_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.487.4794934885.swift)  
[typo_sage_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.487.4794934885.kt)  
[typo_sage_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.487.4794934885.ts)  
[typo_sbermarket_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.487.4794934885.swift)  
[typo_sbermarket_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.487.4794934885.kt)  
[typo_sbermarket_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.487.4794934885.ts)  
[typo_soulmate_ios-swift--canary.487.4794934885.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.487.4794934885.swift)  
[typo_soulmate_kotlin--canary.487.4794934885.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.487.4794934885.kt)  
[typo_soulmate_react-native--canary.487.4794934885.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.487.4794934885.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.205.0-canary.487.4794934885.0
  npm install @salutejs/plasma-core@1.112.0-canary.487.4794934885.0
  npm install @salutejs/plasma-hope@1.205.0-canary.487.4794934885.0
  npm install @salutejs/plasma-icons@1.141.0-canary.487.4794934885.0
  npm install @salutejs/plasma-temple@1.155.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens-b2b@1.17.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens-b2c@0.26.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens-web@1.32.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens@1.48.0-canary.487.4794934885.0
  npm install @salutejs/plasma-ui@1.187.0-canary.487.4794934885.0
  npm install @salutejs/plasma-web@1.205.0-canary.487.4794934885.0
  npm install @salutejs/plasma-cy-utils@0.55.0-canary.487.4794934885.0
  npm install @salutejs/plasma-sb-utils@0.110.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens-android@2.55.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens-ios-swift@2.55.0-canary.487.4794934885.0
  npm install @salutejs/plasma-tokens-utils@0.23.0-canary.487.4794934885.0
  # or 
  yarn add @salutejs/plasma-b2c@1.205.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-core@1.112.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-hope@1.205.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-icons@1.141.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-temple@1.155.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens-b2b@1.17.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens-b2c@0.26.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens-web@1.32.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens@1.48.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-ui@1.187.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-web@1.205.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-cy-utils@0.55.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-sb-utils@0.110.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens-android@2.55.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.55.0-canary.487.4794934885.0
  yarn add @salutejs/plasma-tokens-utils@0.23.0-canary.487.4794934885.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-487/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-487/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-487/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-487/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-487/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-487/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-487/temple-storybook/

File: packages/plasma-ui/src/components/Device/DeviceDetection.tsx [hunk 1]
```diff
@@ -20,7 +20,6 @@ const typoSizes = {
 
 const StandardTypo = createGlobalStyle(standard);
 const CompatibleTypo = createGlobalStyle(compatible);
-const Size = createGlobalStyle(sizes);
```
Comment (Yeti-or) [inline packages/plasma-ui/src/components/Device/DeviceDetection.tsx:23]: типа в токенах больше размеры не лежат?
это же не было публичным API да ?

File: packages/plasma-tokens/generate.ts [hunk 2]
```diff
@@ -62,7 +61,7 @@ writeGeneratedToFS(
 );
 
 // Отдельные файлы для импорта в компонентах
-writeGeneratedToFS(THEMES_VALUES_DIR, generateColorThemeValues({ ...themesColorTokenGroupsFallback, ...colorThemes }));
+writeGeneratedToFS(THEMES_VALUES_DIR, generateColorThemeValues({ ...colorThemes }));
```
Comment (Yeti-or) [inline packages/plasma-tokens/generate.ts]: фолбек для sizes нужен был?

File: packages/plasma-tokens/generate.ts [hunk 3]
```diff
@@ -97,7 +90,7 @@ writeGeneratedToFS(TYPOGRAPHY_DIR, generateTypography(typoSystem.typoStyles));
 writeGeneratedToFS(TYPOGRAPHY_VALUES_DIR, generateTypographyValues(typoSystem));
 
 // Типографика по типам устройств для создания глобального стиля
-writeGeneratedToFS(TYPO_DIR, generateTypoSystem(typo, sizes));
+writeGeneratedToFS(TYPO_DIR, generateTypoSystem(typo));
```
Comment (Yeti-or) [inline packages/plasma-tokens/generate.ts]: типографика для устройств в размерах не нуждалась?

File: packages/plasma-ui/src/components/Button/ActionButton.sizes.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { css } from 'styled-components';
+
+export const sizes = css`
+    --plasma-outline-size: 0.125rem;
```
Comment (Yeti-or) [inline packages/plasma-ui/src/components/Button/ActionButton.sizes.ts:4]: только в кнопках использовался?

File: packages/plasma-tokens/generate.ts [hunk 5]
```diff
@@ -254,75 +251,4 @@ Object.entries(shadows).forEach(([key, type]) => {
...
-//   sizes
-//   width: [],
-//   heights: [],
-//   maxWidths: [],
-//   zIndices
-//   transitions
-//   shadows
-//   gradients: {},
-// }
+// TODO: при добавлении токенов в темы https://theme-ui.com/theme-spec/#styles
```
Comment (Yeti-or) [inline packages/plasma-tokens/generate.ts]: а нам точно надо это TODO ?

File: packages/plasma-tokens/lib/creators/properties/createDirectories.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import fs from 'fs';
```
Comment (Yeti-or) [inline packages/plasma-tokens/lib/creators/properties/createDirectories.ts:1]: Давай https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md#ensuredirdiroptionscallback

File: packages/plasma-tokens/lib/creators/properties/createLegacyTheme.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    ...convertGroupTokens(legacyTokenGroups.speech, theme),
+});
+
+const generateColors = () =>
+    Object.entries(colorThemes).reduce((acc, [themeName, theme]) => {
+        acc[themeName] = convertGroupedTokenData(theme);
+        return acc;
+    }, {} as Record<string, ThemeTokens>);
+
+// TODO: Это генерация устаревших цветовых токенов, необходимо удалить после перехода на новые токены
```
Comment (Yeti-or) [inline packages/plasma-tokens/lib/creators/properties/createLegacyTheme.ts:32]: заведи плиз issue

File: packages/plasma-tokens/lib/creators/properties/createProperties.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+const propsColorBrandsDir = path.join(propsColorDir, 'brands');
+const propsShadowDir = path.join(propsDir, 'shadow');
+const propsTypoDir = path.join(propsDir, 'typo');
+
+export const createProperties = (
+    themesColorTokenGroups: Record<string, ThemeTokenDataGroups>,
+    shadowTokens: Record<string, any>,
+    typoArchetypes: Record<string, any>,
+) => {
+    createDirectories([propsDir, propsColorDir, propsColorBrandsDir, propsShadowDir, propsTypoDir]);
```
Comment (Yeti-or) [inline packages/plasma-tokens/lib/creators/properties/createProperties.ts:24]: и я бы тупо в каждом методе вызывал ensureDir а не завязывался на отдельный вызов подготовки

File: packages/plasma-tokens/lib/tokensGenerator/index.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+export { themesColorTokenGroups, themesColorTokenGroupsFallback } from './generateColorTokens';
+
+export const typoArchetypes = {
+    mage: mageTypoProperties,
+    sage: sageTypoProperties,
+    soulmate: soulmateTypoProperties,
+    ruler: rulerTypoProperties,
+    plasma: plasmaTypoProperties,
+    sbermarket: sbermarketTypoProperties,
```
Comment (Yeti-or) [inline packages/plasma-tokens/lib/tokensGenerator/index.ts:20]: @gibushnev  а у нас идея что не можно типографическую систему тоже кастомизировать? типа не все сидят на [mage / sage / soulmate / ruler ] ??

File: packages/plasma-tokens/lib/creators/brands/createBrands.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import fs from 'fs';
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/brands/createBrands.ts:1]: у нас в этом пакете (plasma-tokens) смесь: темы для plasma-ui (eva,joy,sber,b2b и т.д.), и темы не привязанные к платформе, которые были созданы генератором (sberprime, sbermarket, sberhealth)

File: packages/plasma-ui/src/components/Device/DeviceDetection.tsx (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Device/DeviceDetection.tsx:23]: да. На самом деле теперь даже с новыми схемами работают некоторые компоненты корректно (например те же кнопки). 

Ну этот экспорт наружу торчал, но его не должны были использовать юзеры, т.к. эти значения только для кнопок и вряд-ли они кому-то могли понадобиться

File: packages/plasma-tokens/generate.ts (hunk 2 above)
Comment (neretin-trike) [reply packages/plasma-tokens/generate.ts]: тут не совсем про sizes. Это про то, что теперь в эту папочку `themeValues` не попадают значения для новых схем. Т.к. у нас стоит задача в новой версии плазмы вообще эту директорию снести (сейчас не можем из-за обратной совместимости тем для plasma-ui)

image

File: packages/plasma-tokens/generate.ts (hunk 3 above)
Comment (neretin-trike) [reply packages/plasma-tokens/generate.ts]: теперь нет, т.к. эти значения лежат непосредственно в компонентах Button и ActionButton

File: packages/plasma-ui/src/components/Button/ActionButton.sizes.ts (hunk 4 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Button/ActionButton.sizes.ts:4]: ага, только в кнопках

File: packages/plasma-tokens/generate.ts (hunk 5 above)
Comment (neretin-trike) [reply packages/plasma-tokens/generate.ts]: ну этот TODO точно не нужен как ISSUE, но как заметку я хотел бы его оставить

File: packages/plasma-tokens/lib/tokensGenerator/mapDeprecatedColorTokens.ts [hunk 11]
```diff
@@ -64,16 +64,23 @@ const getThemeTokenDataGroupsByName = (themeTokenDataGroups: Record<string, Them
...
+    // Данный префикс будет удалён на этапе генерации файлов с токенами (values.ts и index.ts)
+    const legacyPrefix = 'plasmaColors';
+
+    return Object.entries(deprecatedColorTokenOnActualToken)
         .filter(([_, actualName]) => tokens[actualName])
         .reduce(
             (acc, [oldName, actualName]) => ({
                 ...acc,
-                [`${oldName}`]: { value: tokens[actualName].value, comment: `@deprecated instead use ${actualName}` },
+                [`${legacyPrefix}-${oldName}`]: {
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/tokensGenerator/mapDeprecatedColorTokens.ts:76]: по-идее ничего. Хочется токены держать без префиксов, т.к. на этом строим все новые схемы и темы

File: packages/plasma-tokens/lib/creators/properties/createDirectories.ts (hunk 6 above)
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/properties/createDirectories.ts:1]: оке, попробую 👍

File: packages/plasma-tokens/lib/creators/brands/createBrands.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/creators/brands/createBrands.ts:1]: хотлось бы это как-то разрулить =(

File: packages/plasma-tokens/generate.ts (hunk 5 above)
Comment (Yeti-or) [reply packages/plasma-tokens/generate.ts]: не совсем понимаю что она теперь нам говорит и что TODO =)

File: packages/plasma-tokens/lib/tokensGenerator/mapDeprecatedColorTokens.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/tokensGenerator/mapDeprecatedColorTokens.ts:76]: вот чувствую придут к нам с тем что у них коллизии )

File: packages/plasma-tokens/lib/creators/brands/createBrands.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/brands/createBrands.ts:1]: угу, но кажется уже только в plasma 2.0 можно будет сделать

File: packages/plasma-tokens/generate.ts (hunk 5 above)
Comment (neretin-trike) [reply packages/plasma-tokens/generate.ts]: ну там пример того, как можно хранить тему в формате JSON со всеми группами. Типа скругления, тени и т.д., можно подсмотреть у них формат

File: packages/plasma-tokens/lib/tokensGenerator/mapDeprecatedColorTokens.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/tokensGenerator/mapDeprecatedColorTokens.ts:76]: вроде не должны, эти префиксы нужны только для того, чтобы старые токены заработали. Если они их не юзали (а они точно не использовали их, т.к. они не работали), то не должны

File: packages/plasma-tokens/lib/creators/properties/createProperties.ts (hunk 8 above)
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/properties/createProperties.ts:24]: в итоге я обнаружил метод `writeGeneratedToFS`, который у нас есть в plasma-tokens-utils, и он внутри себя как раз все эти проверки и создания файлов делает. Но в других местах, где можно было, за использовал `ensureDir`

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-487/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-487/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-487/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-487/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-487/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-487/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-487/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-tokens/lib/creators/brands/createBrands.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
+import fs from 'fs-extra';
```
Comment (Yakutoc) [inline packages/plasma-tokens/lib/creators/brands/createBrands.ts:1]: [Уточнение]

@neretin-trike Подскажи пожалуйста, почему этот пакет, а нет нативный `fs`?

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-tokens/lib/creators/brands/createBrands.ts (hunk 12 above)
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/brands/createBrands.ts:1]: там Вася выше подсказал метод `ensureDir`, который сразу создаёт директорию если её нет (до это у меня было два действия: проверка + создание). И эта утилита есть только в либе `fs-extra`

**Pull request merged**